codegen/smithy-go-codegen/src/main/java/software/amazon/smithy/go/codegen/integration/HttpBindingProtocolGenerator.java

```diff
@@ -975,34 +980,35 @@ private void writeRestDeserializerMember(
 
         switch (binding.getLocation()) {
             case HEADER:
-                writeHeaderDeserializerFunction(writer, model, memberName, targetShape, binding);
+                writeHeaderDeserializerFunction(writer, model, symbolProvider, memberName, targetShape, binding);
                 break;
             case PREFIX_HEADERS:
                 if (!targetShape.isMapShape()) {
                     throw new CodegenException("unexpected prefix-header shape type found in Http bindings");
                 }
-                writePrefixHeaderDeserializerFunction(writer, model, memberName, targetShape, binding);
+                writePrefixHeaderDeserializerFunction(writer, model, symbolProvider, memberName, targetShape, binding);
                 break;
             default:
                 throw new CodegenException("unexpected http binding found");
         }
     }
 
     private void writeHeaderDeserializerFunction(
-            GoWriter writer, Model model, String memberName,
-            Shape targetShape, HttpBinding binding
+            GoWriter writer, Model model, SymbolProvider symbolProvider,
+            String memberName, Shape targetShape, HttpBinding binding
     ) {
         writer.openBlock("if val := response.Header.Get($S); val != $S {", "}",
                 binding.getLocationName(), "", () -> {
-                    String value = generateHttpBindingsValue(writer, model, targetShape, binding, "val");
+                    String value = generateHttpBindingsValue(writer, model, symbolProvider,
+                            targetShape, binding, "val");
                     writer.write("v.$L = $L", memberName,
-                            CodegenUtils.generatePointerReferenceIfPointable(targetShape, value));
+                            CodegenUtils.generatePointerValueIfPointable(writer, targetShape, value));
                 });
     }
 
     private void writePrefixHeaderDeserializerFunction(
-            GoWriter writer, Model model, String memberName,
-            Shape targetShape, HttpBinding binding
+            GoWriter writer, Model model, SymbolProvider symbolProvider,
+            String memberName, Shape targetShape, HttpBinding binding
```

Comment:
nit: Style says that for function parameter definitions that span lines should be formatted such that each parameter is on its own line.

Reply:
cool will update to use that style